libs/nav/src/tpspace/CParameterizedTrajectoryGenerator.cpp

```diff
@@ -477,6 +488,30 @@ void CParameterizedTrajectoryGenerator::CColisionGrid::updateCellInfo(
 	}
 }
 
+/*-----------------------------------------------------------------
+	Updates info into a the CMatrix inside a cell. It stores the 
+	minimum distance of the bot from the obstacle for the paricular
+	(a,d) pair
+------------------------------------------------------------------*/
+
+void CParameterizedTrajectoryGenerator::CClearanceGrid::updateCellInfo(
+	const unsigned int icx,
+	const unsigned int icy,
+	const uint16_t k,
+	const uint16_t n,
+	const float dist)
+{
+	mrpt::math::CMatrix *matrix = cellByIndex(icx, icy);
+	if(!matrix) return;
+
+	//here we determine whether we need to increase size of matrix or not and then resize the matrix
+	const int cols = std::max(static_cast<int>(matrix->cols()), n+1);
+	matrix->resize(m_parent->getAlfaValuesCount(), cols);
```

Comment:
@chatrasen **IMPORTANT** See [this note](http://www.mrpt.org/tutorials/programming/maths-and-geometry/matrices_vectors_arrays_and_linear_algebra_mrpt_and_eigen_classes/#44resizevssetSize) on `resize()`... it will destroy current data!